src/Compilers/Core/Portable/InternalUtilities/YieldAwaitableExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Roslyn.Utilities
+{
+    internal static class YieldAwaitableExtensions
+    {
+        /// <summary>
+        /// Implements <c>ConfigureAwait(bool)</c> for <see cref="Task.Yield"/>. The resulting behavior in asynchronous code
+        /// is the same as one would expect for <see cref="Task.ConfigureAwait(bool)"/>.
+        /// </summary>
+        /// <param name="awaitable">The awaitable provided by <see cref="Task.Yield"/>.</param>
+        /// <param name="continueOnCapturedContext"><inheritdoc cref="Task.ConfigureAwait(bool)"/></param>
+        /// <returns>An object used to await this yield.</returns>
+        public static ConfiguredYieldAwaitable ConfigureAwait(this YieldAwaitable awaitable, bool continueOnCapturedContext)
```

Comment:
Where is this used?

Reply:
It's used by `WaitForExecutingTaskAsync` when `alwaysYield` is true

Reply:
Ah that is subtle. I didn't realize that it returned a different type.